lib/src/status.rs

```diff
@@ -305,18 +308,106 @@ pub(crate) async fn status(opts: super::cli::StatusOpts) -> Result<()> {
     let legacy_opt = if opts.json {
         OutputFormat::Json
     } else {
-        OutputFormat::Yaml
+        if  std::io::stdout().is_terminal() {
+            OutputFormat::HumanReadable
+        } else {
+            OutputFormat::Yaml
+        }
     };
     let format = opts.format.unwrap_or(legacy_opt);
     match format {
         OutputFormat::Json => serde_json::to_writer(&mut out, &host).map_err(anyhow::Error::new),
         OutputFormat::Yaml => serde_yaml::to_writer(&mut out, &host).map_err(anyhow::Error::new),
+        OutputFormat::HumanReadable => human_readable_output(&mut out, &host),  
     }
     .context("Writing to stdout")?;
 
     Ok(())
 }
 
+#[derive(Default, Iterable)]
+struct ImageOpts {
+    version: String,
+    transport: String,
+    digest: String,
+}
+
+fn human_readable_output(mut out: impl Write, host: &Host) -> Result<()> {
+    for (status_string, status) in [
+        ("staged", &host.status.staged),
+        ("booted", &host.status.booted),
+        ("rollback", &host.status.rollback),
+    ] {
+        if let Some(host_status) = status {
+            let mut image_opts: ImageOpts = ImageOpts::default();
+            if let Some(image) = &host_status.image {
+                out.write_all(format!("Current {:?} image: {:?}\n", status_string, image.image.image).as_bytes())?;
+
+                if let Some(version) = &image.version {
+                    image_opts.version = version.to_owned();
+                } else {
+                    image_opts.version = ("No image version defined").to_owned();
+                }
+
+                image_opts.transport = image.image.transport.clone();
+                image_opts.digest = image.image_digest.clone();
+
+                for (field_name, field_value) in image_opts.iter() {
+                    if let Some(field_print) = field_value.downcast_ref::<String>() {
+                        out.write_all(format!("     Image {}: {:?}\n", field_name, field_print).as_bytes())?;
+                    }
```

Comment:
I get why you wanted to do this...but I would say it's somewhat unusual in Rust to do things this way.

I don't think we're saving a *lot* here over something just like:

```
writeln!("    Image version: {}", image.version);
writeln!("    Image transport: {}", image.transport);
writeln!("    Image digest: {}", image.digest);
```
or so